cvs/objects/sectors/include/value_factor_calculator.h

```diff
@@ -0,0 +1,113 @@
+*     and that User is not otherwise prohibited
+* under the Export Laws from receiving the Software.
+* 
+* Copyright 2011 Battelle Memorial Institute.  All Rights Reserved.
+* Distributed as open-source under the terms of the Educational Community 
+* License version 2.0 (ECL 2.0). http://www.opensource.org/licenses/ecl2.php
+* 
+* For further details, see: http://www.globalchange.umd.edu/models/gcam/
+*
+*/
+
+
+#ifndef _VALUE_FACTOR_CALCULATOR_H_
+#define _VALUE_FACTOR_CALCULATOR_H_
+#if defined(_MSC_VER)
+#pragma once
+#endif
+
+/*!
+ * \file value_factor_calculator.h
+ * \ingroup Objects
+ * \brief The ValueFactorCalculator class header file.
+ * \author Matthew Binsted, Matt Mowers
+ */
+
+#include <string>
+#include "sectors/include/ibackup_calculator.h"
+
+
+// Forward declaration
+class IInfo;
+
+/*!
+ * \ingroup Objects
+ * \brief Interface which defines methods for calculating value factor
+ *        of a technology depending on its market share.
+ * \details Defines an interface to an object which determines the value
+ *          factor for a technology. The value factor may use trial values of
+ *          market share. Value factors are specific to electricity sector
+ *          technologies.
+ * \author Matthew Binsted, Matt Mowers
+ */
+class ValueFactorCalculator : public IBackupCalculator {
+public:
+    ValueFactorCalculator();
+    virtual ValueFactorCalculator* clone() const;
+    static const std::string& getXMLNameStatic();
+    virtual const std::string& getXMLName() const;
+    virtual bool isSameType(const std::string& aType) const;
+    virtual const std::string& getName() const;
+    virtual void toDebugXML(const int aPeriod, std::ostream& aOut, Tabs* aTabs) const;
+    virtual void initCalc(const IInfo* aTechInfo);
+
+    virtual double getMarginalBackupCapacity(const std::string& aSector,
+        const std::string& aElectricSector,
+        const std::string& aResource,
+        const std::string& aRegion,
+        const double aTechCapacityFactor,
+        const double aReserveMargin,
+        const double aAverageGridCapacityFactor,
+        const int aPeriod) const;
+
+    virtual double getAverageBackupCapacity(const std::string& aSector,
+        const std::string& aElectricSector,
+        const std::string& aResource,
+        const std::string& aRegion,
+        const double aTechCapacityFactor,
+        const double aReserveMargin,
+        const double aAverageGridCapacityFactor,
+        const int aPeriod) const;
+
+    virtual double getValueFactor(const std::string& aSector,
+        const std::string& aElectricSector,
+        const std::string& aRegion,
+        const int aPeriod) const;
+
+protected:
+        
+    // Define data such that introspection utilities can process the data from this
+    // subclass together with the data members of the parent classes.
+    DEFINE_DATA_WITH_PARENT(
+        IBackupCalculator,
+
+        //! TODO:  leave an explanatory comment
```

Comment:
Do you need to add some explanation still?

Reply:
Yes :)  I'll push this up momentarily.